### PR TITLE
BUG: fix crash while training + update dependency to keras 3

### PR DESCRIPTION
**Pull request by lvjoebro**
fix crashes while training:
- .hdf5 -> .keras (probably due to updated keras version?)
- test set was created with 400k empty records
- training failed due to null classes in category (filter them out more aggressively)

### CODE REVIEW

**Review by theroggy — CHANGES_REQUESTED** [review]
Best add a dependency on keras>=3 to setup.py because with keras <3 the .keras extension probably will give issues.

Also best add an entry to changelog.md

**Review by theroggy — APPROVED** [review]
